### PR TITLE
(Static type-checking fix)  Add `_ScalarMeta(dtype=...)` field.

### PR DESCRIPTION
**Pull request by patrick-kidger**

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -162,6 +162,8 @@ def _dtype(x: Any) -> DType:
 # types return JAX arrays when instantiated.
 
 class _ScalarMeta(type):
+  dtype: Any
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:165]: Can we make this `dtype: np.dtype`?

Comment (patrick-kidger) [reply jax/_src/numpy/lax_numpy.py:165]: Ah, I missed that this was an output, not an input, to `np.dtype`. Will do.

**Review by jakevdp — COMMENTED** [review]
Looks good! If you can squash the changes then we can get this merged.

**Comment by patrick-kidger** [comment]
Done.

**Review by jakevdp — APPROVED** [review]

**Pull request merged**